packages/openactive-broker-microservice/app.js

```diff
@@ -91,28 +98,36 @@ function getRPDE(url, headers, cb) {
         throw `Base URL of RPDE 'next' property ("${getBaseUrl(json.next)}") does not match base URL of RPDE page ("${url}")`;
       }
 
+      context.currentPage = url;
```

Comment:
I think a lot of mistakes and issues can be avoided by never mutating a function's arguments

For example, if you added some logic under the `setTimeout` statement that calls `getRpde()`, and that logic includes an `await`, after the `await`, context may or may not have been mutated by the subsequent calls to `getRpde()`. This may be fine now, but I would say it's quite reasonable for a programmer to not expect that sort of thing to happen, leading to potentially difficult bugs down the road and - if not bugs - more complexity that a programmer has to keep in their mind as to what can happen and what to be careful of.

Therefore, I think we should stick to a rule of never mutating a function's args (I believe the airbnb lint also has a rule for this).

In this case, I would recommend making a new `context` variable that copies the `context` in the function's arguments and adds whatever else needs to be added

Reply:
Yes this was a quick-and-dirty solution. Had thought about maintaining a Map of contexts and having that get updated by this function... will amend so you can see what you think

Reply:
see new `feedContextMap` approach

Reply:
Magic 🧙